packages/plugins/i18n/server/src/services/localizations.ts

```diff
@@ -1,40 +1,49 @@
 import { isEmpty } from 'lodash/fp';
 
-import { async } from '@strapi/utils';
 import { getService } from '../utils';
 
-const isDialectMySQL = () => strapi.db.dialect.client === 'mysql';
-
 /**
  * Update non localized fields of all the related localizations of an entry with the entry values
- * @param {Object} entry entry to update
- * @param {Object} options
- * @param {Object} options.model corresponding model
  */
-const syncNonLocalizedAttributes = async (entry: any, { model }: any) => {
+const syncNonLocalizedAttributes = async (sourceEntry: any, model: any) => {
   const { copyNonLocalizedAttributes } = getService('content-types');
 
-  if (Array.isArray(entry?.localizations)) {
-    const nonLocalizedAttributes = copyNonLocalizedAttributes(model, entry);
-
-    if (isEmpty(nonLocalizedAttributes)) {
-      return;
-    }
-
-    const updateLocalization = (id: any) => {
-      return strapi.entityService.update(model.uid, id, { data: nonLocalizedAttributes });
-    };
-
-    // MySQL/MariaDB can cause deadlocks here if concurrency higher than 1
-    // TODO: use a transaction to avoid deadlocks
-    await async.map(
-      entry.localizations,
-      (localization: any) => updateLocalization(localization.id),
-      {
-        concurrency: isDialectMySQL() && !strapi.db.inTransaction() ? 1 : Infinity,
-      }
-    );
+  const nonLocalizedAttributes = copyNonLocalizedAttributes(model, sourceEntry);
+  if (isEmpty(nonLocalizedAttributes)) {
+    return;
+  }
+
+  const uid = model.uid;
+  const documentId = sourceEntry.documentId;
+  const status = sourceEntry?.publishedAt ? 'published' : 'draft';
```

Comment:
@alexandrebodin similar comment as what we discussed yesterday, publishedAt and locale fields might not be present if users set the fields query.
What if we always force those to be returned?

Reply:
can't we just findOne in db with the fields we need directly so we don't have to enforce this in the repository code

Reply:
makes sense!